client.go

```diff
@@ -514,16 +518,10 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 				dial.Close()
 			}
 		}()
-		defer dial.Close()
 		breader := bufio.NewReader(dial)
 		err = req.Write(dial)
 		if err != nil {
-			select {
-			case <-ctx.Done():
-				return ctx.Err()
-			default:
-				return err
-			}
+			return chooseError(subCtx, err)
```

Comment:
Yup unfortunately tests didn't reveal this so I missed it earlier. Now I realize this is probably a high occurrence in this pattern of making sub-context. Maybe good idea to name the parent context explicitly, a.k.a. `parentCtx`.